src/flint/fq_mat.jl

```diff
@@ -47,11 +47,11 @@ end
   return z
 end
 
-@inline function setindex!(a::FqPolyRepMatrix, u::FqPolyRepFieldElem, i::Int, j::Int)
+@inline function setindex!(a::FqPolyRepMatrix, u::FqPolyRepFieldElemOrPtr, i::Int, j::Int)
   @boundscheck _checkbounds(a, i, j)
-  ccall((:fq_mat_entry_set, libflint), Nothing,
-        (Ref{FqPolyRepMatrix}, Int, Int, Ref{FqPolyRepFieldElem}, Ref{FqPolyRepField}),
-        a, i - 1, j - 1, u, base_ring(a))
+  @ccall libflint.fq_mat_entry_set(
+    a::Ref{FqPolyRepMatrix}, (i-1)::Int, (j-1)::Int, u::Ref{FqPolyRepFieldElem}, base_ring(a)::Ref{FqPolyRepField}
+  )::Nothing
 end
 
 @inline function setindex!(a::FqPolyRepMatrix, u::ZZRingElem, i::Int, j::Int)
```

Comment:
for this function, I would like to make use of `set!` from https://github.com/Nemocas/Nemo.jl/pull/1908 (instead of the ccall).
The problem is that the `set!` implemented there needs to access `parent` of the first arg, so it does not work with pointers there. What do you think about adding a third argument to `set!` that defaults to `parent(first_arg)` (for non-pointers), but in this particular case we can just pass `base_ring(a)`

Reply:
yeah, makes sense. Should the parent be a thread argument, though, or maybe a kwarg (that would make it easier to grep, harder to get confused about?)

Reply:
what do you mean by `thread argument`?

And the way that I understand how kwargs work in julia, a call always involves a NamedTuple construction and another call (to the `kwcall`) function. This may have an overhead that we do not want to have in something low-level as `setindex!`.

Reply:
I vote for making it a normal positional argument.

Reply:
sorry, typo/braino, I meant "third argument"